source/conversion.c

```diff
@@ -0,0 +1,129 @@
+#include "conversion.h"
+#include "draw.h"
+
+#include <png.h>
+
+static void rotate_agbr_counterclockwise(char** bufp, size_t size, size_t width)
+{
+    uint32_t* buf = (uint32_t*)*bufp;
+    uint32_t* out = malloc(size);
+
+    size_t pixel_count = (size/4);
+    size_t height = pixel_count/width;
+
+    for (uint32_t h = 0; h < height; ++h)
```

Comment:
@Jan200101 Roughly how fast does this run? I assume it's pretty fast (despite the nested for), but I can't test it myself.

Reply:
tested on real hardware just to be sure, and the timing of the png and bin path are practically identical.